Application/EdFi.Common/Security/SecureHashAwareSecretVerifier.cs

```diff
@@ -26,7 +29,17 @@ public bool VerifySecret(string key, string presentedSecret, ApiClientSecret act
             }
 
             var actualHash = _packedHashConverter.GetPackedHash(actualSecret.Secret);
-            var presentedHash = _secureHasher.ComputeHash(presentedSecret, actualHash.HashAlgorithm, actualHash.Iterations, actualHash.Salt);
+
+            var hasher = _secureHashers.SingleOrDefault(x => x.GetAlgorithmHashCode == actualHash.HashAlgorithm);
+
+            if (hasher == null)
+            {
+                throw new NotImplementedException(
+                    $"Secure hasher is not found for algorithm indicated by the packed hash => {actualHash.HashAlgorithm}");
+            }
+
+            var presentedHash = hasher
+                .ComputeHash(presentedSecret, actualHash.HashAlgorithm, actualHash.Iterations, actualHash.Salt);
```

Comment:
We have this logic duplicated in two places now, does it make sense to go with a separate interface/class in a Provider pattern here for selecting the appropriate hasher based on a provided AlgorithmHashCode?

Reply:
done came up with IPackedHashProvider as an interface